packages/client-adapters/src/transformers/import-morph-data.js

```diff
@@ -251,16 +251,21 @@ class ImportMorphData {
 
   /**
    * Overrides feature data from an inflection with feature data from the lemma
+   * or other data
    * if required by an engine-specific list of featureTypes
    * @param {String} featureType the feature type name
    * @param {Inflection} inflection the inflection object
    * @param {Lemma[]} lemmas the lemma objects
    */
   overrideInflectionFeatureIfRequired (featureType, inflection, lemmas) {
-    if (this.inflectionOverrides[featureType] &&
-        this.inflectionOverrides[featureType](inflection, lemmas)) {
-      for (const lemma of lemmas.filter(l => l.features[featureType])) {
-        inflection.addFeature(lemma.features[featureType])
+    if (this.inflectionOverrides[featureType]) {
+      const override = this.inflectionOverrides[featureType](inflection, lemmas)
+      if (typeof override === 'boolean' && override) {
+        for (const lemma of lemmas.filter(l => l.features[featureType])) {
+          inflection.addFeature(lemma.features[featureType])
+        }
+      } else {
+        inflection.addFeature(override)

```

Comment:
What concerns me here is that the business logic rely on an implicit agreement about the return value: boolean or a feature. This convention is not documented anywhere. I'm wondering how easy would it be to restore the workflow after several months.

Maybe we should make it more explicit (and easier to document)? Maybe `inflectionOverrides()` should instead return an object like (filed names are just suggestions):
```
{
    overrideWithLemma: (true of false),
    feature:
}
```
With this, it would be easier to see what are the possible return values from the `inflectionOverrides()`. It would also allow us to document that if `overrideWithLemma` is `true`, the `feature` value will not be used. It would also give a possibility to add other types of return values in the future, if there will be a need for that.

What do you think?

Reply:
yes you're right. I was trying to change as little as possible, which is not always the best move. I will incorporate this suggestion.